### ISSUE TITLE
Feature idea: fine control

### ISSUE THREAD

**Issue by SlimeDog** [Rsl1122/Genie#15]
I'm thinking about situational usage. Here's an example:

Normally when in jail, a player cannot do anything other than wait out their sentence. But if they had obtained a lamp (and prudently saved a wish), a useful wish would be "get out of jail free card" (ie. go home or somewhere else, specified somehow). In that situation, I would like to grant that specific wish, but no others.

That kind of scenario would involve testing a number of conditions:
- Is the player in a defined condition (like jail, perhaps derived from Essentials or some other jail mechanism)?
- Is the player in a defined regions (derived from WG or some other region-support plugin)?
- Is the player in a defined biome?
- Did they make an allowable wish (defined in configuration or permissions) for this set of circumstances?
- Maybe other criteria?

Perhaps all of that magic could be provided by regions and permissions, including permissions for specific wishes, coupled to unique (short) identifiers for each wish node with matching permission nodes.

Does it make sense? How would you think about it? Is it doable?

**Comment by AuroraLS3** [comment]
I think the current command wishes might support implementing such wish if there is a command that can unjail a player if run by the console, I'm not sure. 

For condition based wishes the system would require a major overhaul (The whole plugin has been written in ~13 hours at this point, so the code isn't in very good shape in terms of maintainability)

**Comment by SlimeDog** [comment]
Yeah, I understand how this was put together (and congratulations!). My comments were meant to put a little programming worm in your ear. :)
